jdaviz/configs/mosviz/plugins/viewers.py

```diff
@@ -28,23 +29,33 @@ class MosvizImageView(JdavizViewerMixin, BqplotImageView):
 
     def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
+        self.init_astrowidgets_api()
         self._subscribe_to_layers_update()
 
+        self.state.show_axes = False  # Axes are wrong anyway
+        self.figure.fig_margin = {'left': 0, 'bottom': 0, 'top': 0, 'right': 0}
+
     def data(self, cls=None):
         return [layer_state.layer  # .get_object(cls=cls or self.default_class)
                 for layer_state in self.state.layers
                 if hasattr(layer_state, 'layer') and
                 isinstance(layer_state.layer, BaseData)]
 
-    def set_plot_axes(self):
-        self.figure.axes[1].tick_format = None
-        self.figure.axes[0].tick_format = None
+    def _mark_targets(self):
+        table_data = self.jdaviz_app.data_collection['MOS Table']
 
-        self.figure.axes[1].label = "y: pixels"
-        self.figure.axes[0].label = "x: pixels"
+        if ("R.A." not in table_data.component_ids() or
+                "Dec." not in table_data.component_ids()):
+            return
 
-        # Make it so y axis label is not covering tick numbers.
-        self.figure.axes[1].label_offset = "-50"
+        from astropy.coordinates import SkyCoord
+        from astropy.table import QTable
```

Comment:
Depends on the answer to the question above probably, but maybe move these to the top level? I generally dislike imports in functions.

Reply:
Does https://github.com/spacetelescope/jdaviz/pull/2146#discussion_r1170444678 help you making a decision on this?

Reply:
If it's only used in debugging it can stay.

Reply:
The comment doesn't quite give me an answer as to why these imports should be down here. Is there a reason?

Reply:
Because the imports are unnecessary otherwise. If we decide to delete this method one day, it gives a cleaner diff. If you really want the imports to be high up, I can do it too. Lemme know.

Reply:
Understandable, but isn't this directly against PEP8? https://peps.python.org/pep-0008/#imports

> Imports are always put at the top of the file, just after any module comments and docstrings, and before module globals and constants.

Unless those particular imports take a particularly long time, I'd argue following PEP8. I'm personally comfortable with counting this as a value-add rather than a "small debugging thing that might be useful later"

Reply:
Imports moved up.